### ISSUE TITLE
dataset next batch issue.

### ISSUE THREAD

**Issue by tracer9** [jiyanggao/TURN-TAP#1]
Hi Jiyang!
I have been followed your a series of work in action detection area. Very solid work.

Thanks to open source your TURN-TAP model.

When reading your code, I found that in `dataset.py`, you give mini-batch by using:
![image](https://user-images.githubusercontent.com/6302380/30813606-77ea9ed8-a240-11e7-9e78-5b2fd1a76212.png)

The point is: random sample here is not a good choice. when training, some samples may be trained multiple times while others may not be trained. That's werid.

In other words, you cannot train a whole epoch.

I think pytorch's dataset wrapper would be a good choice.

**Comment by jiyanggao** [comment]
Hi, Good idea! Sorry for the late reply. I will try this idea definately.
